### PR TITLE
fix validator and model check during invocation

### PR DESCRIPTION
**Pull request by bentsku**
We had a wrong assumption when using a `RequestValidator` and validating the body of a request. If a method did not have a `requestModel` set, we would reject the request. However, AWS uses the default `Empty` model. We now follow the same logic.

We also did not have validation for validators before, and would return an empty response if no validators existed with that ID. Now, in parity with AWS, it raises an exception instead of returning `None`.
However, that case should not happen because it's not possible to delete an existing `RequestValidator` if it's used by a `Method` in AWS. I'll try to implement this behaviour in following PR. 

We needed to change the code in `invocations.py` to reflect the change when checking if a request is valid.

note: created a `test_apigateway_common`, where we could test "integrations", or how different RestAPI resources work together, on a level above the CRUD one. If you have a better name than "common", I'm all ears 😄 

_fixes in part #7842_

### CODE REVIEW

File: localstack/services/apigateway/invocations.py
Comment (whummer) [inline localstack/services/apigateway/invocations.py:98]: I'm always a bit wary of "should not happen" comments - should we simply remove the `try`/`except` handling in this case, if we think it is not a valid code path?

File: localstack/services/apigateway/provider.py
Comment (whummer) [inline localstack/services/apigateway/provider.py:517]: A wonderful demonstration of the weirdness of API GW patch operations! 😄

File: tests/integration/apigateway/test_apigateway_common.py
Comment (whummer) [inline tests/integration/apigateway/test_apigateway_common.py:93]: nit: for simple integrations, we could try leveraging and extending the `create_rest_api_with_integration` fixture. It currently only supports `dynamodb` and `kinesis` integrations, but Lambda would be a great addition.. 👍

**Review by whummer — APPROVED** [review]
Looks great - as usual, kudos for adding more snapshots to our test suite @bentsku ! 🚀 Added a few minor comments/questions, nothing that should block the merge, though.. 👍

File: localstack/services/apigateway/invocations.py
Comment (bentsku) [reply localstack/services/apigateway/invocations.py:88]: Good idea, will do! 👍

File: localstack/services/apigateway/invocations.py
Comment (bentsku) [reply localstack/services/apigateway/invocations.py:98]: Yes, I think it can happen for now until #7872 is merged. I'd be for letting it be there, I'll mark it with a TODO instead, until we're sure this code path can't happen anymore.

File: tests/integration/apigateway/test_apigateway_common.py
Comment (bentsku) [reply tests/integration/apigateway/test_apigateway_common.py:93]: Makes sense! I need to take a look at it, and how we can customise it. I guess we can always create simple integrations and use `update_*` afterwards, for example the `requestParameters` in that case. There are so many parameters possible.
I'll take a good look a those once I'm starting with integrations. This PR was a bit outside my "normal" work for now, as I was fixing the change in logic now that we are raising exceptions for non-existing resources. 😄

**Pull request merged**
